hack/verify-golint.sh

```diff
@@ -22,19 +22,6 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 
 cd "${KUBE_ROOT}"
 
-GOLINT=${GOLINT:-"golint"}
-PACKAGES=($(go list ./internal/... | grep -v /vendor/))
-bad_files=()
-for package in "${PACKAGES[@]}"; do
-  out=$("${GOLINT}" -min_confidence=0.9 "${package}" | grep -v -E '(should not use dot imports|internal/file/bindata.go)' || :)
-  if [[ -n "${out}" ]]; then
-    bad_files+=("${out}")
-  fi
-done
-if [[ "${#bad_files[@]}" -ne 0 ]]; then
-  echo "!!! '$GOLINT' problems: "
-  echo "${bad_files[@]}"
-  exit 1
-fi
+LINT=${LINT:-golangci-lint}
```

Comment:
Just a headsup on it, we should install golangci-lint if it is not installed to allow users to run it locally.
A go install should do the job, but we need to take care of it (and also depending on where it was downloaded, not commit to the repo!)

Reply:
Yes, I found:
1. For running it locally, I think we can refer to update-codegen.sh: https://github.com/kubernetes/ingress-nginx/blob/ff0cb504d672c2c576d99adcf7ae9abc98857f6d/hack/update-codegen.sh#L24C1-L25C63 to check if golangci-lint exists, and then execute `go install ` or just remind users that they need to download it.
2. And `make static-check` runs the `hack/verify-*.sh` scripts in a container, we can install golangci-lint in that container image: https://github.com/kubernetes/ingress-nginx/blob/main/build/run-in-docker.sh#L47C1-L48C1 or just execute `go install` in the script.